src/eos-updater-apply.c

```diff
@@ -103,14 +111,17 @@ apply (GTask *task,
     goto error;
 
   bootversion = ostree_sysroot_get_bootversion (sysroot);
-  merge_deployment = ostree_sysroot_get_merge_deployment (sysroot, NULL);
+  booted_deployment = eos_updater_get_booted_deployment_from_loaded_sysroot (sysroot,
+                                                                             &error);
+  if (booted_deployment == NULL)
+    goto error;
```

Comment:
I'd avoid mixing `g_autoptr` with `goto`. In this case it might not be a problem since they're both in the same scope, but I have no idea when the destructors run (and if they're anything like C++ destructors, probably at scope closure).